noir/noir-repo/acvm-repo/brillig_vm/src/arithmetic.rs

```diff
@@ -24,17 +20,23 @@ pub(crate) fn evaluate_binary_field_op<F: AcirField>(
     lhs: MemoryValue<F>,
     rhs: MemoryValue<F>,
 ) -> Result<MemoryValue<F>, BrilligArithmeticError> {
-    let MemoryValue::Field(a) = lhs else {
-        return Err(BrilligArithmeticError::MismatchedLhsBitSize {
-            lhs_bit_size: lhs.bit_size(),
-            op_bit_size: F::max_num_bits(),
-        });
+    let a = match lhs {
```

Comment:
Nit: Should be able to keep the `let ... else { .. }` that was deleted for this match statement. The if let is a lot cleaner imo. Should be no reason we cannot have `impl From<BitSize> for u32`

Reply:
In order to implement From<BitSize> for u32 we'd need access to F::max_num_bits() so BitSize would need to be generic on <F: AcirField> and then field generics would start spreading more ):